lib/Sema/BuilderTransform.cpp

```diff
@@ -1736,7 +1736,17 @@ Optional<BraceStmt *> TypeChecker::applyResultBuilderBodyTransform(
 
   // Solve the constraint system.
   SmallVector<Solution, 4> solutions;
-  if (cs.solve(solutions) || solutions.size() != 1) {
+  bool solvingFailed = cs.solve(solutions);
+
+  if (cs.getASTContext().CompletionCallback) {
+    CompletionContextFinder analyzer(func, func->getDeclContext());
+    filterSolutionsForCodeCompletion(solutions, analyzer);
+    for (const auto &solution : solutions) {
+      cs.getASTContext().CompletionCallback->sawSolution(solution);
+    }
```

Comment:
Do we want to just stop here instead of going farther and emitting diagnostics?

Reply:
I’ll try that in a follow-up PR because I’m curious to see if stopping early improves performance and if yes, by how much.

Reply:
I think it makes sense regardless of performance impact because these diagnostics would just get emitted to be discarded.

Reply:
I mostly want to know how big the performance win is. And I wanted to get this one merged to unblock the argument completion PR ;-)

Reply:
Yeah, no worries, I'm not saying it should have been added in this PR, just want to make a point about it being logically correct thing to do :)